src/resolvers/topicResolvers.ts

```diff
@@ -40,10 +40,28 @@ export const Query = {
   },
   async topics(
     _: any,
-    { contentUri }: QueryToTopicsArgs,
+    { contentUri, filterVisible }: QueryToTopicsArgs,
     context: Context,
   ): Promise<GQLTopic[]> {
-    return fetchTopics({ contentUri }, context);
+    const topicList = await fetchTopics({ contentUri }, context);
+    if (!filterVisible) return topicList;
+
+    const topicsWithPath = topicList.filter(t => t.path != null);
+    const subjectList = await Promise.all(
+      topicsWithPath.map(topic => {
+        const subjectId = topic.path.split('/')[1];
+        return fetchSubject(`urn:${subjectId}`, context);
+      }),
```

Comment:
Kunne forsåvidt henta alle subjects i ett kall i stedet for her der det blir feks alt fra ett til kanskje ti kall.

Reply:
Jeg så litt på det, men fant ingen funksjon som kan ta imot noe parametre for å kun hente valgte subjects. For `fetchSubjects(context: Context)` tar kun imot context som parameter og henter alle subjects. Har du noe forslag på åssen det kunne gjøres for å kun hente med gitt `id` eller noe?

Reply:
Nei, du måtte henta alle. Payload blir større, men høgst sannsynligvis færre http kall.